subprojects/docs/src/docs/userguide/dep-man/01-core-dependency-management/dependency_verification.adoc

```diff
@@ -0,0 +1,849 @@
+In particular, you need to be aware of what binaries are brought in transitively and if they are legit.
+To mitigate the security risks and avoid integrating compromised dependencies in your project, Gradle supports _dependency verification_.
+
+Dependency verification consists of two different and complementary operations:
+
+- _checksum verification_, which allows asserting the integrity of a dependency
+- _signature verification_, which allows asserting the provenance of a dependency
+
+Gradle supports both out of the box but performs no dependency verification by default.
+This section will guide you into configuring dependency verification properly for your needs.
+
+This feature can be used for:
+
+- detecting compromised dependencies
+- detecting compromised plugins
+- detecting tampered dependencies in the local dependency caches
+
+[NOTE]
+====
+Dependency verification is an incubating feature: details are subject to change.
+====
+
+[[sub:enabling-verification]]
+== Enabling dependency verification
+
+[[sub:verification-metadata]]
+=== The verification metadata file
+
+[NOTE]
+====
+Currently the only source of dependency verification metadata is this XML configuration file.
+Future versions of Gradle may include other sources (for example via external services).
+====
+
+Dependency verification is automatically enabled once the configuration file for dependency verification is discovered.
+This configuration file is located at `$PROJECT_ROOT/gradle/verification-metadata.xml`.
+This file minimally consists of the following:
+
+[source,xml]
+----
+<?xml version="1.0" encoding="UTF-8"?>
+<verification-metadata>
+   <configuration>
+      <verify-metadata>true</verify-metadata>
+      <verify-signatures>false</verify-signatures>
+    </configuration>
+</verification-metadata>
+----
+
+Doing so, Gradle will verify all artifacts using <<#sec:checksum-verification,checksums>>, but will not verify <<#sec:signature-verification,signatures>>.
+Gradle will verify any artifact downloaded using its dependency management engine, which includes, but is not limited to:
+
+- artifact files (e.g jar files, zips, ...) used during a build
+- metadata artifacts (POM files, Ivy descriptors, Gradle Module Metadata)
+- plugins (both project and settings plugins)
+- artifacts resolved using the advanced dependency resolution APIs
+
+Gradle will _not_ verify changing dependencies (in particular `SNAPSHOT` dependencies) nor locally produced artifacts (typically jars produced during the build itself) as by nature their checksums and signatures would always change.
+
+With such a minimal configuration file, a project using _any_ external dependency or plugin would immediately start failing because it doesn't contain any checksum to verify.
+
+A dependency verification configuration is _global_: a single file is used to configure verification of the whole build.
```

Comment:
Would you please add clarification here re `buildSrc`? E.g. `buildSrc requires its own configuration file` or `buildSrc configuration file`.

Reply:
It's the same file for `buildSrc`